torch/csrc/jit/tensorexpr/expr.h

```diff
@@ -158,6 +158,66 @@ class Var : public ExprNode<Var> {
   std::string name_hint_;
 };
 
+class TORCH_API Buf : public ExprNode<Buf> {
+ public:
+  static ExprHandle make(
+      const std::string& name_hint,
+      const std::vector<ExprHandle>& dims);
+  static ExprHandle make(const std::vector<ExprHandle>& dims);
+
+  // TODO: unique_name
+  const Var* base_handle() const {
+    return base_handle_;
+  }
+  const std::string& name_hint() const {
+    return base_handle_->name_hint();
+  }
+
+  Buf(const Var* var, const std::vector<const Expr*>& dims)
+      : ExprNodeBase(kHandle, kPrimitive), base_handle_(var), dims_(dims) {
+    TORCH_CHECK(var);
+  }
+
+  int ndim() const {
+    return dims_.size();
+  }
+  const Expr* dim(int index) const {
+    return dims_[index];
+  }
+  std::vector<const Expr*> dims() const {
+    return dims_;
+  }
+
+ private:
+  const Var* base_handle_;
+  std::vector<const Expr*> dims_;
+};
+
+class TORCH_API BufHandle : public ExprHandle {
+ public:
+  BufHandle() : ExprHandle(nullptr) {}
+  //   explicit BufHandle(Dtype dtype) : ExprHandle(Buf::make(dtype)) {}
+  BufHandle(const std::string& name_hint, const std::vector<ExprHandle>& dims)
```

Comment:
Should BufHandle carry the dtype to its content?

Reply:
Yes, eventually it should. In this PR I only made `Buf` := `Var` + `dims` (and dtype of the var is `kHandle`), but in follow-ups I plan to add dtype and then we might be able to merge it with `Buffer`.